test/tests/pbs_admin_suspend.py

```diff
@@ -405,3 +413,416 @@ def test_degraded_resv(self):
+        j = Job(TEST_USER)
+        jid1 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid1)
+
+        # admin suspend and resume job in a loop
+        count = 0
+        while count <= 15:
+            self.server.sigjob(jid1, 'admin-suspend', runas=ROOT_USER)
+            self.server.expect(JOB, {'job_state': 'S'}, id=jid1)
+            self.server.expect(NODE, {'state': 'maintenance'}, id='vn[0]')
+
+            # sleep for sometime
+            time.sleep(3)
+
+            # resume the job
+            self.server.sigjob(jid1, 'admin-resume', runas=ROOT_USER)
+            self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+            self.server.expect(NODE, {'state': 'free'}, id='vn[0]')
+
+            # increase the count
+            count += 1
+
+    def test_custom_res(self):
+        """
+        Test that job will not run on a node in
+        maintenance state if explicitly asking
+        for a resource on that node
+        """
+
+        # create multiple vnodes
+        a = {'resources_available.ncpus': 4, 'resources_available.mem': '4gb'}
+        self.server.create_vnodes('vn', a, 3, self.mom, usenatvnode=True)
+
+        # create a node level resource
+        self.server.manager(
+            MGR_CMD_CREATE, RSC, {'type': 'float', 'flag': 'nh'}, id="foo",
+            runas=ROOT_USER)
+
+        # set foo on vn[1]
+        self.server.manager(
+            MGR_CMD_SET, NODE, {'resources_available.foo': 5}, id='vn[1]',
+            runas=ROOT_USER)
+
+        # set foo in sched_config
+        self.scheduler.add_resource('foo')
+
+        # submit a few jobs
+        j = Job(TEST_USER)
+        j.set_attributes({'Resource_List.select': 'vnode=vn[1]'})
+        jid1 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid1)
+
+        # admin suspend the job to put the node to maintenance
+        self.server.sigjob(jid1, 'admin-suspend', runas=ROOT_USER)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid1)
+        self.server.expect(NODE, {'state': 'maintenance'}, id='vn[1]')
+
+        # submit other jobs asking for specific resources on vn[1]
+        j = Job(TEST_USER)
+        j.set_attributes({'foo': '2'})
+        jid2 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid2)
+
+        # submit more jobs. They should be running
+        j = Job(TEST_USER)
+        jid3 = self.server.submit(j)
+        jid4 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid3)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid4)
+
+        # verify that vn[1] is still in maintenance and
+        # job3 and job4 not running on vn[1]
+        self.server.expect(NODE, {'state': 'maintenance'}, id='vn[1]')
+        try:
+            self.server.expect(
+                JOB, {'exec_vnode': 'vn[1]'}, id=jid3, max_attempts=20)
```

Comment:
If you are checking for an exact match, this will fail.  The exec_vnode's format is (vn[1]:ncpus=1:...).  If it does a substring match, then you're fine.  I don't know.